lib/kube/proxy/forwarder.go

```diff
@@ -1427,6 +1428,10 @@ func (f *Forwarder) newClusterSessionRemoteCluster(ctx authContext) (*clusterSes
 		return nil, trace.AccessDenied("access denied: failed to authenticate with auth server")
 	}
 
+	// Set 'kube.teleport.cluster.local' as ServerName to leverage TLS Routing.
+	// The connection will be either forwarded directly to kubernetes_service or by TLS Routing router.
+	tlsConfig.ServerName = fmt.Sprintf("%s.%s", alpnproxy.KubeSNIPrefix, constants.APIDomain)
```

Comment:
Should this only be set if TLS routing is enabled? Maybe we can propagate a config flag to forwarder from service/service.

Reply:
We are dialling  leaf `KubeAddr` on remote leaf cluster by `remote.kube.proxy.teleport.cluster.local` reverse tunnel subsystem. So we don't know here is leaf cluster supports TLS Routing.

cluster  cert is always generated for the `*.teleport.cluster.local` SANS for [Proxy and Kube roles ](https://github.com/gravitational/teleport/blob/master/lib/auth/auth.go#L2095)thus dialling with ServerName `kube.teleport.cluster.local` should be safe.

Before for remote cluster the ServerName was set to `constants.APIDomain` https://github.com/gravitational/teleport/blob/master/lib/kube/proxy/forwarder.go#L1114